include/deal.II/base/tensor_function_parser.h

```diff
@@ -0,0 +1,336 @@
+ * It is necessary to tell the parser that there is an additional variable
+ * to be taken into account (t).
+ * TensorFunctionParser<2,2> tfp;
+ * tfp.initialize(variables,
+ *               expression,
+ *               constants,
+ *               true); // flag for time dependence
+ *
+ * // Point at which we want to evaluate the function
+ * Point<2> point(0.0, 1.0);
+ *
+ * // evaluate the expression at 'point':
+ * double result = tfp.value(point);
+ *
+ * deallog << "Function '" << expression << "'"
+ *         << " @ " << point
+ *         << " is "
+ *         << std::endl
+ *         << result[0][0] << " " << result[0][1]
+ *         << result[1][0] << " " << result[1][1]
+ *         << std::endl;
+ * @endcode
+ *
+ * See also the documentation of the FunctionParser class.
+ *
+ * This class overloads the virtual method value() and value_list() of the
+ * TensorFunction base class with the byte compiled versions of the expressions
+ * given to the initialize() methods. Note that the class will not work unless
+ * you first call the initialize() method that accepts the text description of
+ * the function as an argument (among other things).
+ *
+ * The syntax to describe a function follows usual programming practice, and
+ * is explained in detail at the homepage of the underlying muparser library
+ * at http://muparser.beltoforion.de/ .
+ *
+ *
+ * Vector-valued functions can either be declared using strings where the
+ * function components are separated by semicolons, or using a vector of
+ * strings each defining one vector component.
+ *
+ *
+ * @ingroup functions
+ * @author Konrad Simon, 2019
+ */
+template <int rank, int dim, typename Number = double>
+class TensorFunctionParser : public TensorFunction<rank, dim, Number>
+{
+public:
+  /**
+   * Standard constructor. Only set initial time. This object needs to be
+   * initialized with the initialize() method before you can use it. If an
+   * attempt to use this function is made before the initialize() method has
+   * been called, then an exception is thrown.
+   */
+  TensorFunctionParser(const double initial_time = 0.0);
+
+  /**
+   * Constructor for parsed functions. This object needs to be initialized
+   * with the initialize() method before you can use it. If an attempt to
+   * use this function is made before the initialize() method has been called,
+   * then an exception is thrown.
+   * Takes a semicolon separated list of expressions (one for each component
+   * of the tensor function), an optional comma-separated list of constants.
+   */
+  TensorFunctionParser(
+    const std::string &expression,
+    const std::string &constants      = "",
+    const std::string &variable_names = default_variable_names() + ",t");
+
+  /**
+   * Destructor. Explicitly delete the TensorFunctionParser objects (there is
+   * one for each component of the function).
+   */
+  ~TensorFunctionParser() override;
```

Comment:
We generally explicitly mark it as `virtual`, even if it's implied.

Reply:
Done. Also in function_parser.h.

Reply:
Ah, thanks for looking there as well!